craft_application/application.py

```diff
@@ -519,24 +527,37 @@ def run(  # noqa: PLR0912,PLR0915  (too many branches, too many statements)
 
             provider_name = command.provider_name(dispatcher.parsed_args())
 
-            craft_cli.emit.debug(
-                f"Build plan: platform={platform}, build_for={build_for}"
-            )
-            self._pre_run(dispatcher)
+            # Check that pro services are correctly configured:
+            # Validate requested pro services if we are running in destructive mode ...
+            if not run_managed and not is_managed:
+                craft_cli.emit.debug(
+                    f"Validating requested Ubuntu Pro status on host: {pro_services}"
+                )
+                pro_services.validate()
```

Comment:
Thinking about this some more I feel like this specific bit (validating the requested pro services against the running environment) needs to be done in both cases (destructive and managed mode)

Reply:
> this feels like a lot of work to be doing here when we don't even ...

Yeah that is definitely a problem. If `list-extensions` is not run in a managed instance, the `--pro` arguments will have to match the system, otherwise an exception will be raised. 

Since the `--pro` argument is only parsed in the lifecycle command, I can set the default in `pro_services = getattr(...` to None, then skip validation whenever the default is used. This should limit the validation to lifecycle commands.

There is probably better place to put this validation code so the run method doesn't get cluttered. I'll wait for Alex to comment first.

> Thinking about this some more I feel like this specific bit ...

I agree, Anas and I were talking about this earlier to validate pro services when re entering a instance. Some of the resolution strings in the Exceptions raised by ProServices will have change because they only make sense in destructive mode.
https://github.com/clay-lake/craft-application/blob/78bdfb9db4be3a2eb04f69adde4663b41d0da374/craft_application/errors.py#L253-L254

Reply:
@clay-lake we discussed this work yesterday and the decision is to go ahead with the implementation; we'll review it but while it's on this prototype phase we won't block it unless it's something critical

Reply:
I agree this is probably something to do on the lifecycle command itself (but also non-blocking here).